sample_sandy.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import os
+import sys
+import tarfile
+from multiprocessing import Pool
+import re
+import shutil
+from pathlib import Path
+
+import openmc.data
+
+
+description = """
+This scripts generates random (gaussian) evaluations of a nuclear data file following 
+its covariance matrix using SANDY, and converts them to HDF5 for use in OpenMC. Script
+generates a cross_sections_sandy.xml file with the standard library plus the sampled evaluations.
+"""
+
+
+class CustomFormatter(
+    argparse.ArgumentDefaultsHelpFormatter, argparse.RawDescriptionHelpFormatter
+):
+    pass
+
+
+parser = argparse.ArgumentParser(
+    description=description, formatter_class=CustomFormatter)
+
+parser.add_argument("-n", "--nuclides", nargs="+", 
+                    default="Fe56", help="The nuclide(s) to be sampled")
+parser.add_argument("-d", "--destination", default=None, 
+                    help="Directory to create new library in")
+parser.add_argument("-l", "--libdir", default=None, 
+                    help="Directory of endf library to sample")
+parser.add_argument("-s", "--samples", default=200, 
+                    help="Number of samples per nuclide")
+parser.add_argument("-p", "--processes", default=1, 
+                    help="number of worker processes (default = 1)")
+parser.add_argument("-f", "--format_only", default=False,
+                    help="Only format previously sampled files to HDF5")
+
+args = parser.parse_args()
+script_dir = Path.cwd()
+
+output_dir = args.destination
+if output_dir == None:
+    output_dir = script_dir / "sandy_rand"
+else:
+    output_dir = Path(output_dir).resolve()
+
+endf_files_dir = output_dir / "endf"
+hdf5_files_dir = output_dir / "hdf5"
+
+libdir = args.libdir
+if libdir == None:
+    libdir = Path(os.getenv("NUCLEAR_DATA_DIR")) / "nndc-b7.1-endf"
```

Comment:
Personally, I would just make libdir a required argument rather than an optional argument that uses this (undocumented) environment variable as a fallback